test/hardhat.go

```diff
@@ -0,0 +1,84 @@
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type HardhatTest struct {
+	t          *testing.T
+	checkFuncs []func(string, error)
+	setupFuncs []func() error
+	projectDir string
+}
+
+func NewHardhatTest(t *testing.T, projectDir string) *HardhatTest {
+	return &HardhatTest{
+		t:          t,
+		projectDir: projectDir,
+	}
+}
+
+func (h *HardhatTest) NPMInstall() *HardhatTest {
+	setupFunc := func() error {
+		cmd := exec.Command("npm", "install")
+		cmd.Dir = h.projectDir
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+		h.t.Logf("Running command: '%s'", cmd.String())
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("failed to install npm deps: %v", err)
+		}
+		return nil
+	}
+	h.setupFuncs = append(h.setupFuncs, setupFunc)
+	return h
+}
+
+func (h *HardhatTest) NoError() *HardhatTest {
+	checkFunc := func(_ string, err error) {
+		assert.NoError(h.t, err)
+	}
+	h.checkFuncs = append(h.checkFuncs, checkFunc)
+	return h
+}
```

Comment:
Maybe this is not needed and instead we always want to assert that `npm hardhat test` succeeded. On the other hand, this would allow us to test that misconfigurations fail. Additionally, we can use `OutputContains` to check the specific reason for failure